components-api/src/main/scala/pl/touk/nussknacker/engine/api/typed/TypeConversionHandler.scala

```diff
@@ -86,6 +96,35 @@ object TypeConversionHandler {
       handleLooseNumberConversion(boxedGivenClass, boxedSuperclassCandidate)
   }
 
+  private def handleIndexedRecordToMapConversion(
+      from: SingleTypingResult,
+      to: TypedClass,
+      strict: Boolean
+  ): Boolean = {
+    if (!AssignabilityUtil.isAssignableToLoadableClass(
+        from.runtimeObjType.klass,
+        "org.apache.avro.generic.IndexedRecord"
+      ) ||
+      !ClassUtils.isAssignable(to.runtimeObjType.klass, classOf[java.util.Map[_, _]])) {
+      false
+    } else {
+      val indexedRecordKeyParam = Typed.genericTypeClass(classOf[String], List())
```

Comment:
1. Just `Typed[String]`, `String` is not a generic type with zero list of parameters. We should probably have `Nel` instead of `List` for parameters.
2. Inline this, name of the variable doesn't help. `IndexedRecord` doesn't have key generic parameter